### PR TITLE
Release/v2.0.0

### PR DESCRIPTION
**Pull request by santidrj**

### CODE REVIEW

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Coverage decreased per file
===========================
- src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoImpl.java  -6
         

Complexity increasing per file
==============================
- src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoImpl.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5561436&bid=18238099)

File: src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoImpl.java [hunk 1]
```diff
@@ -179,7 +180,11 @@ public void deleteGroupSubscription(String userUid, String groupName, WriteBatch
 
     @Override
     public void saveMessagingToken(String uid, String token) throws DatabaseAccessException, DocumentException {
+        String currentToken = documentAdapter.getStringFromDocument(Path.ofDocument(Collections.users, uid), "FCM");
```
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoImpl.java:183]: Codacy Issue found: [The String "FCM" appears 4 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5561436)

File: src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoImpl.java [hunk 2]
```diff
@@ -467,4 +472,28 @@ private void throwExceptionIfUserDoesNotExist(DocumentSnapshot userDoc) throws D
...
+        throws DatabaseAccessException, DocumentException {
+        ApiFuture<QuerySnapshot> subscribedGroups = collectionAdapter
+            .getDocumentsWhereArrayContains(Path.ofCollection(Collections.groups), "notification-tokens", currentToken);
+        try {
+            for (QueryDocumentSnapshot group : subscribedGroups.get().getDocuments()) {
+                batch.update(group.getReference(), "FCM", FieldValue.arrayRemove(currentToken), "FCM",
+                    FieldValue.arrayUnion(token));
+            }
+        } catch (InterruptedException e) {
+            throw new DatabaseAccessException("write-failed", e.getMessage());
```
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoImpl.java:494]: Codacy Issue found: [The String "write-failed" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5561436)

**Review by marclll — DISMISSED** [review]

**Review by oricat8 — DISMISSED** [review]

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Coverage decreased per file
===========================
- src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoImpl.java  -6
         

Complexity increasing per file
==============================
- src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoImpl.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5561436&bid=18238099)

File: src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoImpl.java (hunk 1 above)
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoImpl.java:183]: Codacy Issue found: [The String "FCM" appears 4 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5561436)

File: src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoImpl.java (hunk 2 above)
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoImpl.java:494]: Codacy Issue found: [The String "write-failed" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5561436)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoImpl.java  1
         

Complexity decreasing per file
==============================
+ src/main/java/org/pesmypetcare/webservice/builders/Path.java  -4
+ src/main/java/org/pesmypetcare/webservice/builders/CollectionPath.java  -4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5561436&bid=18238099)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoImpl.java  1
         

Complexity decreasing per file
==============================
+ src/main/java/org/pesmypetcare/webservice/builders/Path.java  -4
+ src/main/java/org/pesmypetcare/webservice/builders/CollectionPath.java  -4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5561436&bid=18238099)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoImpl.java  1
         

Complexity decreasing per file
==============================
+ src/main/java/org/pesmypetcare/webservice/builders/Path.java  -4
+ src/main/java/org/pesmypetcare/webservice/builders/CollectionPath.java  -4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5561436&bid=18238099)

**Review by marclll — DISMISSED** [review]
Ok

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage decreased per file
===========================
- src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoImpl.java  -6
         

Complexity increasing per file
==============================
- src/main/java/org/pesmypetcare/webservice/dao/usermanager/UserDaoImpl.java  1
         

Complexity decreasing per file
==============================
+ src/main/java/org/pesmypetcare/webservice/builders/Path.java  -4
+ src/main/java/org/pesmypetcare/webservice/builders/CollectionPath.java  -4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5561436&bid=18238099)

**Review by oricat8 — APPROVED** [review]

**Review by marclll — APPROVED** [review]

**Pull request merged**